tests/check_linux_layout_intel.py

```diff
@@ -0,0 +1,192 @@
+#
+# Copyright (c) 2024, Intel Corporation.
+
+import argparse
+import os
+import pathlib
+
+sof_ri_info = None
+try:
+    import sof_ri_info
+except ImportError:
+    print("Note: not using sof_ri_info")
+    pass
+
+# List of Intel firmware files and their paths the upstream SOF Linux driver
+# expects to find under /lib/firmware/intel. A sof-bin release should have
+# binaries for all these targets.
+fw_list = [
+    "sof/sof-apl.ri",
+    "sof/sof-adl-n.ri",
+    "sof/sof-ehl.ri",
+    "sof/sof-glk.ri",
+    "sof/sof-adl.ri",
+    "sof/sof-rpl.ri",
+    "sof/community/sof-apl.ri",
+    "sof/community/sof-adl-n.ri",
+    "sof/community/sof-ehl.ri",
+    "sof/community/sof-glk.ri",
+    "sof/community/sof-adl.ri",
+    "sof/community/sof-rpl.ri",
+    "sof/community/sof-cnl.ri",
+    "sof/community/sof-tgl.ri",
+    "sof/community/sof-icl.ri",
+    "sof/community/sof-tgl-h.ri",
+    "sof/community/sof-rpl-s.ri",
+    "sof/community/sof-jsl.ri",
+    "sof/community/sof-adl-s.ri",
+    "sof/community/sof-cfl.ri",
+    "sof/community/sof-cml.ri",
+    "sof/intel-signed/sof-apl.ri",
+    "sof/intel-signed/sof-adl-n.ri",
+    "sof/intel-signed/sof-ehl.ri",
+    "sof/intel-signed/sof-glk.ri",
+    "sof/intel-signed/sof-adl.ri",
+    "sof/intel-signed/sof-rpl.ri",
+    "sof/intel-signed/sof-cnl.ri",
+    "sof/intel-signed/sof-tgl.ri",
+    "sof/intel-signed/sof-icl.ri",
+    "sof/intel-signed/sof-tgl-h.ri",
+    "sof/intel-signed/sof-rpl-s.ri",
+    "sof/intel-signed/sof-jsl.ri",
+    "sof/intel-signed/sof-adl-s.ri",
+    "sof/intel-signed/sof-cfl.ri",
+    "sof/intel-signed/sof-cml.ri",
+    "sof/sof-cnl.ri",
+    "sof/sof-cht.ri",
+    "sof/sof-tgl.ri",
+    "sof/sof-byt.ri",
+    "sof/sof-icl.ri",
+    "sof/sof-bdw.ri",
+    "sof/sof-tgl-h.ri",
+    "sof/sof-rpl-s.ri",
+    "sof/sof-jsl.ri",
+    "sof/sof-adl-s.ri",
+    "sof/sof-cfl.ri",
+    "sof/sof-cml.ri",
+    "sof-ipc4/rpl/sof-rpl.ri",
+    "sof-ipc4/rpl/community/sof-rpl.ri",
+    "sof-ipc4/rpl/intel-signed/sof-rpl.ri",
+    "sof-ipc4/adl-n/sof-adl-n.ri",
+    "sof-ipc4/adl-n/community/sof-adl-n.ri",
+    "sof-ipc4/adl-n/intel-signed/sof-adl-n.ri",
+    "sof-ipc4/tgl-h/community/sof-tgl-h.ri",
+    "sof-ipc4/tgl-h/intel-signed/sof-tgl-h.ri",
+    "sof-ipc4/tgl-h/sof-tgl-h.ri",
+    "sof-ipc4/mtl/community/sof-mtl.ri",
+    "sof-ipc4/mtl/intel-signed/sof-mtl.ri",
+    "sof-ipc4/mtl/sof-mtl.ri",
+    "sof-ipc4/rpl-s/community/sof-rpl-s.ri",
+    "sof-ipc4/rpl-s/intel-signed/sof-rpl-s.ri",
+    "sof-ipc4/rpl-s/sof-rpl-s.ri",
+    "sof-ipc4/adl-s/community/sof-adl-s.ri",
+    "sof-ipc4/adl-s/intel-signed/sof-adl-s.ri",
+    "sof-ipc4/adl-s/sof-adl-s.ri",
+    "sof-ipc4/tgl/community/sof-tgl.ri",
+    "sof-ipc4/tgl/intel-signed/sof-tgl.ri",
+    "sof-ipc4/tgl/sof-tgl.ri",
+    "sof-ipc4/adl/sof-adl.ri",
+    "sof-ipc4/adl/community/sof-adl.ri",
+    "sof-ipc4/adl/intel-signed/sof-adl.ri",
+ ]
+
+BASE="/lib/firmware/intel"
+
+def check_fw_files_kernel_to_bundle():
+    for i in fw_list:
```

Comment:
Python seems overkill for this. Also, it's a too "static" and inconvenient to have the test data and test code in the same file. Maybe not in the short term but it should be possible to "test" old and new releases with the latest test code.

I think it would be much simpler to use the underrated `tree` and `diff` commands for this one-liner:

```shell
diff -u expected_tree.txt <(cd release_candidate && tree -naF)
```

BTW `tarball_multi_releases.bash` produces a manifest, maybe that could be an alternative approach?

Reply:
Thanks for feedback @marc-hb ! This may indeed be overkill, but I wanted to codify the checks I'm doing manually and I felt more comfortable writing this in python. The "static" property is something intentional as we will regularly have updates to "expected_tree.txt" (e.g. new binaries to be included in next SOF2.10 release) that I'd like to track explicitly in version control. If you want to test older sof-bin, you take check_linux_layout_intel.py from that version of sof-bin.

But but, maybe this is overkill. Certainly the expected manifest could be stored separately and be e.g. tree output or tarball_multi_releases manifest.

Reply:
>  If you want to test older sof-bin, you take check_linux_layout_intel.py from that version of sof-bin.

Older binaries are **not** versioned in git; _they stay in the main branch_. Sometimes "old" binaries are even added later, a good example is ce5cb4e2d93d. 

This gives the flexibility to make a change to test code like check_linux_layout_intel.py and "test the test code" against a _range_ of older binaries and releases.

> we will regularly have updates to "expected_tree.txt" (e.g. new binaries to be included in next SOF2.10 release)

There could/should be one "expected_tree_2024_3.txt" data file per release present in the main branch, just like old binaries are present in the main branch.

Even if not or not yet, pulling such data out of git history should not require changing the python test code which is probably being edited.

So, at a very minimum please separate data and code into different files.